kedro/contrib/io/networkx_local/networkx_local.py

```diff
@@ -0,0 +1,109 @@
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""``NetworkXLocalDataSet`` loads and saves graphs to a local json file in node/link format using
+``NetworkX``.
+See https://networkx.github.io/documentation/stable/tutorial.html for details.
+"""
+
+import json
+from os.path import isfile
+from typing import Any, Dict, Union
+
+import networkx
+
+from kedro.contrib.io import DefaultArgumentsMixIn
+from kedro.io import AbstractDataSet
+
+
+class NetworkXLocalDataSet(DefaultArgumentsMixIn, AbstractDataSet):
+    """
+    ``NetworkXLocalDataSet`` loads and saves graphs to a local json file in node/link format using
+    ``NetworkX``.
+    See https://networkx.github.io/documentation/stable/tutorial.html for details.
+    Example:
+    ::
+        >>> from kedro.contrib.io.networkx import NetworkXLocalDataSet
+        >>> graph = NetworkXLocalDataSet(filepath="graph.json")
+        >>> bb = networkx.betweenness_centrality(graph)
+        >>> nx.set_node_attributes(graph, bb, 'betweenness')
+        >>> graph.save()
+        >>> reloaded = graph.load()
+        >>> assert graph == reloaded
+    """
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``NetworkXLocalDataSet``.
+
+        Args:
+            filepath: The path to the NetworkX graph JSON file.
+
+            load_args: Arguments passed on to ```networkx.node_link_graph``.
+                See https://networkx.github.io/documentation for details.
+
+            save_args: Arguments passed on to ```networkx.node_link_data``.
+                See https://networkx.github.io/documentation for details.
+
+        """
+        self._filepath = filepath
+        super().__init__(load_args, save_args)
+
+    def _load(self) -> Union[networkx.Graph, Dict[str, networkx.Graph]]:
+        graph = None
+        with open(self._filepath, "r") as input_file:
+            json_payload = json.loads(input_file.readline())
+            graph = networkx.node_link_graph(json_payload, **self._load_args)
+
+        return graph
+
+    def _save(self, data: networkx.Graph) -> None:
+        json_graph = networkx.node_link_data(data)
+        target_path = (
+            self._save_args["output_filepath"]
```

Comment:
When do users need to pass `output_filepath` in _save_args . rather than `self._filepath`?

Reply:
My mistake; I like to save versions of the graph as I go the process of enriching the properties, but it looks like that can be done in catalog.yml without `output_filepath`